index.js

```diff
@@ -45,9 +44,488 @@ app.intent('Default Welcome Intent', conv => {
+});
+
+/**
+ * add.participant.to.group - yes
+ */
+app.intent('add.participant.to.group - yes', async conv => {
+  const circuit = await getCircuit(conv);
+  if (!circuit) {
+    return;
+  }
+
+  const { users, convs } = conv.contexts.input['addparticipantgroup_data'].parameters;
+  const thisUser = circuit.user;
+
+  if (convs[0].participants.indexOf(users[0].userId) !== -1) {
+    // User is already in the conversation
+    conv.ask(`${users[0].displayName} is already a participant in ${convs[0].topic}. Is there anything else I can do for you?`, new Suggestions('Yes', 'No'));
+    conv.contexts.delete('addparticipantgroup_data');
+    conv.contexts.set('anything_else', 2);
+    return;
+  }
+
+  if (convs[0].isModerated && convs[0].moderators.indexOf(thisUser.userId) === -1) {
+    // Conversation is moderated and user is not a moderator
+    conv.ask(`Sorry, but you are not a moderator in ${convs[0].topic} so ${users[0].displayName} cannot be added. Is there anything else I can do for you?`, new Suggestions('Yes', 'No'));
+    conv.contexts.delete('addparticipantgroup_data');
+    conv.contexts.set('anything_else', 2);
+    return;
+  }
+
+  await circuit.addParticipant(convs[0].convId, users[0].userId, true);
+  conv.ask(`${users[0].displayName} was added to ${convs[0].topic}. Is there anything else I can do for you?`, new Suggestions('Yes', 'No'));
+  conv.contexts.delete('addparticipantgroup_data');
+  conv.contexts.set('anything_else', 2);
+});
+
+/**
+ * add.participant.to.one
+ */
+app.intent('add.participant.to.one', async (conv, { thirdUser, target }) => {
+  const circuit = await getCircuit(conv);
+  if (!circuit) {
+    return;
+  }
+
+  // Save results to context
+  conv.contexts.set('addparticipantone_data', 5, {
+    thirdUsers: thirdUsers,
+    targetUsers: targetUsers
+  });
+
+  let thirdUsers = await searchUsers(circuit, conv, thirdUser);
+
+  if (thirdUsers.length !== 1) {
+    conv.contexts.set('addparticipantone_getuser', 5);
+    return;
+  }
+
+  let targetUsers = await searchUsers(circuit, conv, target);
+
+  if (targetUsers.length !== 1) {
+    conv.contexts.set('addparticipantone_getuser', 5);
+    return;
+  }
+
+  // One result found for thirdUser and target
+  conv.ask(`Ready to add ${thirdUsers[0].displayName} to your conversation with ${targetUsers[0].displayName}?`, new Suggestions('Yes', 'No'));
+});
+
+/**
+ * add.participant.to.one - collect.user
+ */
+app.intent('add.participant.to.one - collect.user', async conv => {
+  const circuit = await getCircuit(conv);
+  if (!circuit) {
+    return;
+  }
+
+  let { thirdUsers, targetUsers } = conv.contexts.input['addparticipantone_data'].parameters;
+  let user = conv.parameters.user;
+  let users = await searchUsers(circuit, conv, user);
+
+  if (users.length !== 1) {
+    // Stays in this intent and gets input from user again
+    return;
+  }
+
+  // One user found beyond this point
+  let params = conv.contexts.input['addparticipantone_data'].parameters;
```

Comment:
const

Reply:
Actually why do you keep reassigning params?

Reply:
To save the updated values to the context whenever necessary. I did still change it from <b>let</b> to <b>const</b> since only the object's properties are changing.